sbin/spark-daemon.sh

```diff
@@ -121,45 +121,63 @@ if [ "$SPARK_NICENESS" = "" ]; then
     export SPARK_NICENESS=0
 fi
 
+run_command() {
+  mode=$1
+  shift
 
-case $option in
-
-  (start|spark-submit)
-
-    mkdir -p "$SPARK_PID_DIR"
+  mkdir -p "$SPARK_PID_DIR"
 
-    if [ -f $pid ]; then
-      TARGET_ID="$(cat "$pid")"
-      if [[ $(ps -p "$TARGET_ID" -o args=) =~ $command ]]; then
-        echo "$command running as process $TARGET_ID.  Stop it first."
-        exit 1
-      fi
+  if [ -f $pid ]; then
+    TARGET_ID="$(cat "$pid")"
+    if [[ $(ps -p "$TARGET_ID" -o args=) =~ $command ]]; then
+      echo "$command running as process $TARGET_ID.  Stop it first."
+      exit 1
     fi
+  fi
 
-    if [ "$SPARK_MASTER" != "" ]; then
-      echo rsync from "$SPARK_MASTER"
-      rsync -a -e ssh --delete --exclude=.svn --exclude='logs/*' --exclude='contrib/hod/logs/*' $SPARK_MASTER/ "$SPARK_HOME"
-    fi
+  if [ "$SPARK_MASTER" != "" ]; then
+    echo rsync from "$SPARK_MASTER"
+    rsync -a -e ssh --delete --exclude=.svn --exclude='logs/*' --exclude='contrib/hod/logs/*' $SPARK_MASTER/ "$SPARK_HOME"
```

Comment:
```
"$SPARK_MASTER/"
```

Unless this variable should potentially be interpreted as multiple arguments to `rsync`.